dev-util/sysdig: renovate & bump to latest versions

hhoffstaette opened the pull request:
This updates sysdig to 0.29.3, the current release. Our version in the tree is 0.27.1 and severely outdated; since then the project has changed its structure and updating is overdue, otherwise we will just fall behind even further.

This PR contains:

- an update to sysdig itself
- a new version of the driver, now called scap-driver (instead of sysdig-kmod); this is purely to avoid confusion and be consistent with the main application
- a new (header-only) dependency called valijson. I went to great lengths to make this a separate package (including tests); alternatively we can also just stuff it into the main ebuild as build-only dependency.

The main complication for all this is that - despite the theoretical separation of sysdig and its libs - the build does *not* currently build with released versions of the falcosecurity/libs package. The main app's build pulls the libs from a checked/verified git tag, *not* from a semver released package. The libs also cannot be built as separate ebuild, since they are static-only, unversioned and the main sysdig build does not properly reference its own libs when they are prebuilt or outside the tree. So for now we need to make sure that driver and main app reference the same falcosecurity/libs tag. I wish this was easier, but it's not (and under heavy construction upstream), so this is the best we can do for now while retaining the split between sysdig & the driver.

thesamesam commented on dev-cpp/valijson/metadata.xml:
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "https://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+	<!-- maintainer-needed -->
```
We can't add new packages as m-n: https://projects.gentoo.org/qa/policy-guide/maintainer.html#pg0602.

Also, I know you've been avoiding it (😆) but I think you're the best person to handle sysdig & friends, even if it's a difficult package to handle sometimes, as you actually use it.

hhoffstaette replied:
Added myself as proxy-maintainer.

thesamesam commented on dev-cpp/valijson/valijson-0.6.ebuild:
```diff
@@ -0,0 +1,45 @@
+HOMEPAGE="https://github.com/tristanpenman/valijson"
+SRC_URI="https://github.com/tristanpenman/valijson/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+IUSE="test"
```
Put "IUSE" after "KEYWORDS" (follow order from skel.ebuild)

hhoffstaette replied:
done

thesamesam commented on dev-cpp/valijson/valijson-0.6.ebuild:
```diff
@@ -0,0 +1,45 @@
+
+IUSE="test"
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc ~ppc64 ~riscv ~x86"
+LICENSE="BSD-2"
```
Resort to: LICENSE/SLOT/KEYWORDS/IUSE/RESTRICT.

hhoffstaette replied:
done

thesamesam commented on dev-cpp/valijson/valijson-0.6.ebuild:
```diff
@@ -0,0 +1,45 @@
+	if use test; then
+		sed -i -e "s:../tests/data/documents/:../${P}/tests/data/documents/:" tests/test_adapter_comparison.cpp || die
+		sed -i -e "s:../tests/data:../${P}/tests/data:" tests/test_validation_errors.cpp || die
+		sed -i -e "s:../thirdparty/:../${P}/thirdparty/:" tests/test_validator.cpp || die
```
You can combine these two into one sed call with -e (same file).

hhoffstaette replied:
done

thesamesam commented on dev-cpp/valijson/valijson-0.6.ebuild:
```diff
@@ -0,0 +1,45 @@
+	cmake_src_configure
+}
+
+src_compile() {
```
You don't need this?

hhoffstaette replied:
removed

thesamesam commented on dev-cpp/valijson/valijson-0.6.ebuild:
```diff
@@ -0,0 +1,45 @@
+	doheader -r include/*
+}
+
+src_test() {
```
Move this above src_install to reflect phase execution order.

hhoffstaette replied:
done

thesamesam commented on dev-util/scap-driver/scap-driver-0.29.3.ebuild:
```diff
@@ -0,0 +1,63 @@
+
+S="${WORKDIR}/libs-${COMMIT}"
+
+pkg_pretend() {
```
cmake doesn't export a pkg_pretend, so you can drop this and it'll have the same effect.

hhoffstaette replied:
removed

thesamesam commented on dev-util/scap-driver/scap-driver-0.29.3.ebuild:
```diff
@@ -0,0 +1,63 @@
+	linux-mod_pkg_setup
+}
+
+pkg_setup() {
```
Drop as per other comment.

thesamesam commented on dev-util/scap-driver/scap-driver-0.29.3.ebuild:
```diff
@@ -0,0 +1,63 @@
+}
+
+src_prepare() {
+	cmake_src_prepare
```
Drop as per other comment.

thesamesam commented on dev-util/scap-driver/scap-driver-0.29.3.ebuild:
```diff
@@ -0,0 +1,63 @@
+
+src_configure() {
+	local mycmakeargs=(
+		# we will use linux-mod, so just pretend to use bundled deps
```
good, thanks for explaining

thesamesam commented on dev-util/scap-driver/scap-driver-0.29.3.ebuild:
```diff
@@ -0,0 +1,63 @@
+	local mycmakeargs=(
+		# we will use linux-mod, so just pretend to use bundled deps
+		# in order to make it through the cmake setup
+		-DUSE_BUNDLED_DEPS=on
```
Nit: We usually do ON and OFF, not on and off.

hhoffstaette replied:
All changed to ON/OFF

thesamesam commented on dev-util/sysdig/sysdig-0.29.3.ebuild:
```diff
@@ -0,0 +1,88 @@
+REQUIRED_USE="${LUA_REQUIRED_USE}"
+
+RDEPEND="${LUA_DEPS}
+	app-misc/jq:0=
```
1. Don't put an explicit 0 unless it has another slot
2. Only do := if it exposes a subslot for its ABI and we link against it (e.g. curl doesn't have one)
3. Sort deps

hhoffstaette replied:
I checked all deps via `equery meta <pkg>` and hopefuly got them all right.

thesamesam commented on dev-util/sysdig/sysdig-0.29.3.ebuild:
```diff
@@ -0,0 +1,88 @@
+src_configure() {
+	local mycmakeargs=(
+		# don't build driver
+		-DBUILD_DRIVER=off
```
```suggestion
		-DBUILD_DRIVER=OFF
```

hhoffstaette replied:
done

thesamesam commented on dev-util/sysdig/sysdig-0.29.3.ebuild:
```diff
@@ -0,0 +1,88 @@
+		-DBUILD_LIBSCAP_EXAMPLES=OFF
+
+		# point to the falcosecurity-libs tree
+		-DFALCOSECURITY_LIBS_SOURCE_DIR=${WORKDIR}/libs-${LIBS_COMMIT}
```
```suggestion
		-DFALCOSECURITY_LIBS_SOURCE_DIR="${WORKDIR}"/libs-${LIBS_COMMIT}
```

hhoffstaette replied:
done

thesamesam commented on dev-util/sysdig/sysdig-0.29.3.ebuild:
```diff
@@ -0,0 +1,88 @@
+		-DUSE_BUNDLED_DEPS=OFF
+
+		# add valijson include path to prevent downloading
+		-DVALIJSON_INCLUDE=/usr/include
```
Maybe:
```suggestion
		-DVALIJSON_INCLUDE="${ESYSROOT}"/usr/include
```

hhoffstaette replied:
done

thesamesam reviewed (changes requested):
Thanks!

hhoffstaette commented:
Thanks so much for your suggestions! I'll go over them in the next few days. As for maintainer-needed, I can certainly be proxy-maintainer for these ebuilds.

thesamesam commented:
I knew I'd get you!

hhoffstaette commented:
I addressed all your points and also added a few more comments esp. in scap-driver about why we're using a git revision for the libs dependency. Also set myself as proxy-maintainer in all three packages, even though buildbot now complains that a real maintainer is missing..? In any case not sure what else I can do here.

A final note on arches for sysdig: it's currently limited to ~amd/~x86 because of libb64's keywords, which only in the tree for sysdig; the other two I set to the major arches where everything should work (~amd64, ~arm64, x86).

thesamesam commented:
Thanks! I guess we should last-rite -kmod in due course?

hhoffstaette commented:
> Thanks! I guess we should last-rite -kmod in due course?

Both sysdig-0.27 and -kmod, yes. They're over 2 years old by now.
